sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/DataSourceCodecTest.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources
+
+import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.test.SQLTestUtils
+
+abstract class DataSourceCodecTest extends QueryTest with SQLTestUtils {
```

Comment:
If we are going to test only file-based data source, we can make a single simple suite like `FileBasedDataSourceSuite`.

Reply:
Maybe, `FileBasedDataSourceCodecSuite`?
```scala
  private val allFileBasedDataSources = Seq("orc", "parquet", ...)

  allFileBasedDataSources.foreach { format =>
    test(s"... - $format") {
```

Reply:
ok, sounds good. let me refactor it.

Reply:
Actually, this code style is a bit hard to extend (e.g. how to test avro?) and I was planning to refactor the existing test suites as well.

I think a better solution is
```
trait FileSourceCodecSuite ...

class ParquetCodecSuite extends FileSourceCodecSuite

class OrcCodecSuite ...
```

Reply:
@cloud-fan's idea is more close to what I have in mind at the beginning. @dongjoon-hyun is it good for you too?

Reply:
Note: I think these test suites can be in one file if possible.

Reply:
Yes, @viirya and @cloud-fan , +1 for the proposed structure in a single file.